packages/dx-react-scheduler/src/plugins/drag-drop-provider.jsx

```diff
@@ -0,0 +1,322 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import moment from 'moment';
+import {
+  Plugin, Template, TemplatePlaceholder,
+  TemplateConnector, DropTarget, DragSource,
+  DragDropProvider as DragDropProviderCore,
+} from '@devexpress/dx-react-core';
+import {
+  cellIndex,
+  cellData,
+  cellType,
+  allDayRects,
+  verticalTimeTableRects,
+  horizontalTimeTableRects,
+  getAppointmentStyle,
+  intervalDuration,
+  VERTICAL_TYPE,
+  SCROLL_OFFSET,
+  SCROLL_SPEED_PX,
+  SECONDS,
+} from '@devexpress/dx-scheduler-core';
+
+const pluginDependencies = [
+  { name: 'EditingState' },
+  { name: 'Appointments' },
+  { name: 'DayView', optional: true },
+  { name: 'WeekView', optional: true },
+  { name: 'MonthView', optional: true },
+  { name: 'AllDayPanel', optional: true },
+];
+
+export class DragDropProvider extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      startTime: null,
+      endTime: null,
+      payload: undefined,
+    };
+
+    this.resetCache = this.resetCache.bind(this);
+    this.onOver = (getters, actions) => args => this.handleOver.bind(this)(args, getters, actions);
```

Comment:
From the way these three functions are used they can be declared as common class methods.